### PR TITLE
Upgrade to Realm JavaScript v11.3.0-rc.1

### PR DESCRIPTION
**Pull request by kneth**
Migrating to [`flatted`](https://www.npmjs.com/package/flatted) for exporting to JSON.

The option `--legacy-peer-deps` was added to help `npm ci` to resolve a conflict between Studio and https://github.com/realm/react-realm-context (likely because RJS v11.3.0-rc.1 is a prerelease).

### CODE REVIEW

File: .github/workflows/prs-lint-build-package.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       with:
        node-version: '16'
     - name: Install dependencies
-      run: npm ci
+      run: npm ci --legacy-peer-deps
```
Comment (takameyer) [inline .github/workflows/prs-lint-build-package.yml:19]: We can probably remove this post release

**Review by takameyer — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,35 @@
 # Changelog
 
-All notable changes to this project will be documented in this file.
```
Comment (bmunkholm) [inline CHANGELOG.md:3]: why delete this?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
-This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+## vNext
```
Comment (bmunkholm) [inline CHANGELOG.md:3]: Is version updated later?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,35 @@
...
+
+- Upgraded Realm JS to v11.3.0-rc.1.
+
+### Compatibility
+
+Use this version of Realm Studio to read and write Realm database files, using the same lock file format as:
+
+| SDK              | Version          |
+| ---------------- | ---------------- |
+| Realm JavaScript | v11.3.0-rc.0 - ? |
```
Comment (bmunkholm) [inline CHANGELOG.md]: You will update RJS to v11.3.0 and thats the one people should use, right? (even if it's not released as of the release of Studio)

File: CHANGELOG.md (hunk 3 above)
Comment (kneth) [reply CHANGELOG.md:3]: It will be part of the release process. This PR is only an update of a dependency.

File: CHANGELOG.md (hunk 4 above)
Comment (kneth) [reply CHANGELOG.md]: We can release 13.0.1 based on RJS v11.3.0 when it is out

File: CHANGELOG.md (hunk 2 above)
Comment (kneth) [reply CHANGELOG.md:3]: See my comment below

File: CHANGELOG.md (hunk 4 above)
Comment (bmunkholm) [reply CHANGELOG.md]: What I meant was that in this list we should just reference 11.3.0. They shouldn't use the rc.0 version of RJS.
I don't think there is a need for us to make yet another Studio release and bother people with upgrading if there is no code changes.

**Pull request merged**